internal/statemachine/common_states.go

```diff
@@ -345,7 +345,7 @@ func (stateMachine *StateMachine) makeDisk() error {
 	// TODO: this is only temporarily needed until go-diskfs is fixed - see below
 	var existingDiskIds [][]byte
 	for volumeName, volume := range stateMachine.GadgetInfo.Volumes {
-		imgName := filepath.Join(stateMachine.commonFlags.OutputDir, volumeName+".img")
+		imgName := filepath.Join(stateMachine.commonFlags.OutputDir, stateMachine.VolumeNames[volumeName])
```

Comment:
What is not completely clear for me is: are we creating `makeDisk()` disks for every volume still, or only for volumes that have been explicitly requested by the `artifacts:` field? Looks to me like we still iterate over the gadget set of volumes. I think it's best if we don't do something the image-definition file doesn't want: if there is more than one volume defined in the gadget but we only define one artifact output, we should only create the disk image for that one volume. This is important as at some point we might support more than just raw `.img` files, and users might want to create one volume as a raw image, second as a qcow2 image etc.